broker/tasks/alb.py

```diff
@@ -25,6 +31,68 @@ def get_lowest_used_alb(listener_arns):
     listener_data = listener_data["Listeners"][0]
     return listener_data["LoadBalancerArn"], listener_data["ListenerArn"]
 
+
+def load_albs(dedicated_listener_arns):
+    stmt = insert(DedicatedALBListener).values(
+        [dict(listener_arn=arn) for arn in dedicated_listener_arns]
+    )
+    stmt = stmt.on_conflict_do_nothing(index_elements=["listener_arn"])
+    db.session.execute(stmt)
+
+
+def get_lowest_dedicated_alb(service_instance, db):
+    # TODO: handle grabbing the next available ALB when ours are full, or nearly so
+    potential_listeners = DedicatedALBListener.query.filter(
+        DedicatedALBListener.dedicated_org == service_instance.org_id
+    ).all()
+    if len(potential_listeners) == 0:
+        potential_listeners = DedicatedALBListener.query.filter(
+            DedicatedALBListener.dedicated_org == None
+        ).all()
+    arns = [listener.listener_arn for listener in potential_listeners]
+    arns.sort()  # this just makes testing easier
+    alb_arn, alb_listener_arn = get_lowest_used_alb(arns)
+    selected_listener = [
+        listener
+        for listener in potential_listeners
+        if listener.listener_arn == alb_listener_arn
+    ][0]
+    selected_listener.alb_arn = alb_arn
+    selected_listener.dedicated_org = service_instance.org_id
+
+    service_instance.alb_arn = alb_arn
+    service_instance.alb_listener_arn = alb_listener_arn
+
+    db.session.add(service_instance)
+    db.session.add(selected_listener)
+    db.session.commit()
+
+
+@huey.nonretriable_task
+def load_albs_from_config():
+    load_albs(config.DEDICATED_ALB_LISTENER_ARNS)
+
+
+@huey.retriable_task
+def select_dedicated_alb(operation_id, **kwargs):
+    operation = Operation.query.get(operation_id)
+    service_instance = operation.service_instance
+
+    operation.step_description = "Selecting load balancer"
+    flag_modified(operation, "step_description")
+    db.session.add(operation)
+    db.session.commit()
+
+    if (
+        service_instance.alb_arn
+        and operation.action == Operation.Actions.PROVISION.value
+    ):
+        return
```

Comment:
Why do we just return on provision? I thought this is where we would need to select the lowest dedicated ALB for the domain?

Reply:
this is an idempotency catch - if we're provisioning and we already have an alb_arn, we already did this step and can jump along to the next one. 

These types of checks are for when we re-enqueue a pipeline that got orphaned